doc_src/index.hdr.in

```diff
@@ -849,12 +849,14 @@ The user can change the settings of `fish` by changing the values of certain var
 
 - `FISH_VERSION`, the version of the currently running fish
 
+- `SHLVL`, the level of nesting of shells
+
+These two variables are not read-only and can be set by the user. They will be used if the `stty` rows and cols values are zero (which can happen on a serial console). Fish checks that the values re reasonable which means cols >= 20 and rows >= 2. Note that fish will also modify these vars to match what it believes to be the size of the terminal. This means that the terminal size reported by the kernel has precedence over these variables.
```

Comment:
I like it. But I've tweaked your verbiage to make it clear they must both be set. The way I've deliberately structured the code it won't use one if the other isn't set.